extensions/amp-analytics/0.1/events.js

```diff
@@ -1468,6 +1472,29 @@ export class VisibilityTracker extends EventTracker {
       );
     }
 
+    const getUnlistenPromiseForSelector = (selector, selectionMethod) => {
+      return this.root
+        .getAmpElement(
+          context.parentElement || context,
+          selector,
+          selectionMethod
+        )
+        .then(element => {
+          return visibilityManagerPromise.then(
+            visibilityManager => {
+              return visibilityManager.listenElement(
+                element,
+                visibilitySpec,
+                this.getReadyPromise(waitForSpec, selector, element),
+                createReportReadyPromiseFunc,
+                this.onEvent_.bind(this, eventType, listener, element)
+              );
+            },
+            () => {}
+          );
+        });
+    };
+
     let unlistenPromise;
     // Root selectors are delegated to analytics roots.
     if (!selector || selector == ':root' || selector == ':host') {
```

Comment:
What's the plan for special case `selector = [':host']`

Reply:
I think for these special cases (`!selector || selector == ':root' || selector == ':host'`), we should leave default functionality and not alter behavior.

Reply:
So `selector = [':host']` doesn't work?

Reply:
Yes, it does not work. Should we account for this situation?

Reply:
IMO `selector = [':host', '#hero-image']` is a valid use case that track the pageview and the visibility of the hero image. However one can also argue that pageview is different from all visibility event that it needs its own configuration. 

It's not immediately blocking the PR, but the code structure may look very different if we decide to support the feature.

Reply:
Yes I agree in the future it might call for a refactor, but for now I'm ok with not including `:host` in the arrays, until the need arises.